instrumentation/http/src/main/java/brave/http/HttpRequest.java

```diff
@@ -24,21 +26,63 @@
  * @since 5.8
  */
 public abstract class HttpRequest extends Request {
-  /** @see HttpAdapter#startTimestamp(Object) */
+  /**
+   * The timestamp in epoch microseconds of the beginning of this request or zero to take this
+   * implicitly from the current clock. Defaults to zero.
+   *
+   * <p>This is helpful in two scenarios: late parsing and avoiding redundant timestamp overhead.
+   * If a server span, this helps reach the "original" beginning of the request, which is always
+   * prior to parsing.
+   *
+   * <p>Note: Overriding has the same problems as using {@link brave.Span#start(long)}. For
+   * example, it can result in negative duration if the clock used is allowed to correct backwards.
+   * It can also result in misalignments in the trace, unless {@link brave.Tracing.Builder#clock(Clock)}
+   * uses the same implementation.
+   *
+   * @see HttpResponse#finishTimestamp()
+   * @see brave.Span#start(long)
+   * @see brave.Tracing#clock(TraceContext)
+   * @since 5.8
+   */
   public long startTimestamp() {
     return 0L;
   }
 
-  /** @see HttpAdapter#method(Object) */
+  /**
+   * The HTTP method, or verb, such as "GET" or "POST" or null if unreadable.
+   *
+   * <p>Conventionally associated with the key "http.method"
+   *
+   * <h3>Note</h3>
+   * <p>It is part of the <a href="https://tools.ietf.org/html/rfc7231#section-4.1">HTTP RFC</a>
+   * that an HTTP method is case-sensitive. Do not downcase results. If you do, not only will
+   * integration tests fail, but you will surprise any consumers who expect compliant results.
+   */
   @Nullable public abstract String method();
```

Comment:
It's probably too late now due to backwards compatibility, but I guess we didn't need to use `@Nullable` on attributes that are directly part of the HTTP spec. e.g., an `HttpRequest` without a `method` never exists and doesn't make sense. I find the business logic hard to follow because of the null checks that end up scattered around.

Reply:
agree on this.. throwing when method is null would be new, but we could remove the nullable

Reply:
https://github.com/openzipkin/brave/issues/1089